azure-pipelines.yml

```diff
@@ -1,44 +1,76 @@
-variables: {
-  MODULE_VERSION: '1.2.2',
-  NODE_8: '8.x',
-  NODE_10: '10.x',
-  NODE_12: '12.x',
-  NODE_14: '14.x'
-}
+variables:
+    { MODULE_VERSION: "1.2.2", NODE_10: "10.x", NODE_12: "12.x", NODE_14: "14.x", NODE_16: "16.x" }
 name: $(MODULE_VERSION)-$(Date:yyyyMMdd)$(Rev:.r)
 
 trigger:
-- main
-- dev
+    branches:
+        include:
+            - main
+            - dev
 
 jobs:
-- job: UnitTests
-  strategy:
-    matrix:
-      WINDOWS_NODE8:
-        IMAGE_TYPE: 'vs2017-win2016'
-        NODE_VERSION: $(NODE_8)
-      WINDOWS_NODE10:
-        IMAGE_TYPE: 'vs2017-win2016'
-        NODE_VERSION: $(NODE_10)
-      WINDOWS_NODE12:
-        IMAGE_TYPE: 'vs2017-win2016'
-        NODE_VERSION: $(NODE_12)
-      WINDOWS_NODE14:
-        IMAGE_TYPE: 'vs2017-win2016'
-        NODE_VERSION: $(NODE_14)
-  pool:
-    vmImage: $(IMAGE_TYPE)
-  steps:
-  - task: NodeTool@0
-    inputs:
-      versionSpec: $(NODE_VERSION)
-    displayName: 'Install Node.js'
-  - script: npm install
-    displayName: 'npm install'
-  - script: npm run test
-    displayName: 'npm build and test'
-    condition: ne(variables['NODE_VERSION'], variables['NODE_8'])
-  - script: npm run test:nolint
-    displayName: 'npm build and test (no linting)'
-    condition: eq(variables['NODE_VERSION'], variables['NODE_8'])
+    - job: UnitTests
+      strategy:
+          matrix:
+              UBUNTU_NODE10:
+                  IMAGE_TYPE: "ubuntu-latest"
+                  NODE_VERSION: $(NODE_10)
+              UBUNTU_NODE14:

```

Comment:
Are you intentionally skipping NODE12?

Reply:
Seems I skipped it because it wasn't in the current `azure-pipelines.yml` of the nodeJS worker :)
I'll add it here, but just a heads up:  https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/azure-pipelines.yml

Reply:
You can see our version matrix here: https://github.com/Azure/azure-functions-nodejs-worker#azure-functions-nodejs-worker

We support 14/16 on the latest worker, but the previous worker supports 10/12/14. Since you included 10, I assumed you were supporting both the latest worker and previous worker

Reply:
I see. I don't see a reason why we wouldn't include Node12 if we have Node10, so we'll test them both.